System/Posix/Files/Common.hsc

```diff
@@ -688,3 +753,418 @@ getFdPathVar (Fd fd) v =
+fileDaxX _ = error "fileDaxX not available on this platform"
+#endif
+
+fileBlockSizeX (ExtendedFileStatus statx) = unsafePerformIO $ do
+  r <- withForeignPtr statx $ (#peek struct statx, stx_blksize) :: IO Word32
+  return $ CBlkSize (fromIntegral r)
+fileAttributesX (ExtendedFileStatus statx) =
+  unsafePerformIO $ withForeignPtr statx $ (#peek struct statx, stx_attributes)
+fileLinkCountX (ExtendedFileStatus statx) = unsafePerformIO $ do
+  links <- withForeignPtr statx $ (#peek struct statx, stx_nlink) :: IO Word32
+  return $ CNlink (fromIntegral links)
+fileOwnerX (ExtendedFileStatus statx) =
+  unsafePerformIO $ withForeignPtr statx $ (#peek struct statx, stx_uid)
+fileGroupX (ExtendedFileStatus statx) =
+  unsafePerformIO $ withForeignPtr statx $ (#peek struct statx, stx_gid)
+fileModeX (ExtendedFileStatus statx) = unsafePerformIO $ do
+  r <- withForeignPtr statx $ (#peek struct statx, stx_mode) :: IO Word16
+  return $ CMode $ fromIntegral r
+fileIDX (ExtendedFileStatus statx) =
+  unsafePerformIO $ withForeignPtr statx $ (#peek struct statx, stx_ino)
+fileSizeX (ExtendedFileStatus statx) =
+  unsafePerformIO $ withForeignPtr statx $ (#peek struct statx, stx_size) :: Word64
+fileBlocksX (ExtendedFileStatus statx) =
+  unsafePerformIO $ withForeignPtr statx $ (#peek struct statx, stx_blocks) :: Word64
+fileAttributesMaskX (ExtendedFileStatus statx) =
+  unsafePerformIO $ withForeignPtr statx $ (#peek struct statx, stx_attributes_mask)
+fileAccessTimeHiResX (ExtendedFileStatus statx) =
+  unsafePerformIO $ withForeignPtr statx $ \statx_ptr -> do
+    sec  <- (#peek struct statx, stx_atime.tv_sec) statx_ptr :: IO EpochTime
+    nsec <- (#peek struct statx, stx_atime.tv_nsec) statx_ptr :: IO (#type int)
+    return $ timeHiResToNominalDiffTime sec nsec
+fileCreationTimeHiResX (ExtendedFileStatus statx) =
+  unsafePerformIO $ withForeignPtr statx $ \statx_ptr -> do
+    sec  <- (#peek struct statx, stx_btime.tv_sec) statx_ptr :: IO EpochTime
+    nsec <- (#peek struct statx, stx_btime.tv_nsec) statx_ptr :: IO (#type int)
+    return $ timeHiResToNominalDiffTime sec nsec
+fileStatusChangeTimeHiResX (ExtendedFileStatus statx) =
+  unsafePerformIO $ withForeignPtr statx $ \statx_ptr -> do
+    sec  <- (#peek struct statx, stx_ctime.tv_sec) statx_ptr :: IO EpochTime
+    nsec <- (#peek struct statx, stx_ctime.tv_nsec) statx_ptr :: IO (#type int)
+    return $ timeHiResToNominalDiffTime sec nsec
+fileModificationTimeHiResX (ExtendedFileStatus statx) =
+  unsafePerformIO $ withForeignPtr statx $ \statx_ptr -> do
+    sec  <- (#peek struct statx, stx_mtime.tv_sec) statx_ptr :: IO EpochTime
+    nsec <- (#peek struct statx, stx_mtime.tv_nsec) statx_ptr :: IO (#type int)
+    return $ timeHiResToNominalDiffTime sec nsec
+
+timeHiResToNominalDiffTime :: EpochTime -> Int32 -> POSIXTime
+timeHiResToNominalDiffTime (CTime sec) nsec = realToFrac $ picosecondsToDiffTime $ toInteger sec * 1e12 + toInteger nsec * 1e3
```

Comment:
`secondsToNominalDiffTime` is not available on older `time` versions, so have to resort to `picosecondsToDiffTime` and `realToFrac` (which has a rewrite rule between DiffTime and NominalDiffTime)

Reply:
I think it's time we drop support for GHC 8.2 and 8.4, I have not seen anyone using them for ages.

Reply:
Done